management-api-server/src/main/java/com/datastax/mgmtapi/resources/NodeOpsResources.java

```diff
@@ -355,8 +356,7 @@ public Response getStreamInfo() {
       content =
           @Content(
               mediaType = MediaType.APPLICATION_JSON,
-              schema = @Schema(implementation = String.class),
-              examples = @ExampleObject(value = SNAPSHOT_DETAILS_RESPONSE_EXAMPLE)))
+              schema = @Schema(implementation = SnapshotDetails.class)))
   @Operation(summary = "Retrieve snapshot details", operationId = "getSnapshotDetails")
   public Response getSnapshotDetails(
       @QueryParam("snapshotNames") List<String> snapshotNames,
```

Comment:
We need a version of this method which takes no arguments. The current JMX implementation appears to simply return all snapshots. The parameters can be optional if that is easier/possible.

Reply:
Looking into the OpenAPI code generator, it doesn't support any kind of Optional query parameter markings. The client code that is generated does take a list for both parameters (snapshot names and keyspaces), but calling the method with `null` will result in the client simply omitting the parameters. So in Reaper, you would do something like this:
```java
SnapshotDetails details = apiClient.getSnapshotDetails(null, null);
```
which would make the query to Management API and not provide either parameter. On the Management API side here, it interprets `null` as "get all of them".

Reply:
Can we use the Optional<T> type? https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html

Reply:
At least then it is clear what the behaviour will be from the type system, if not in the docs via some annotation.

Reply:
OK, I can see the issue with generating the 4 methods. I can also see that if we switch the types to Optional<List> for both parms then that's a breaking change, which we don't necessarily want to sign up to now. For v2 perhaps we can get better typing on the parameters so that methods 2 and 3 are distinguishable?

For now, could we get java doc on the method to explain this?

Reply:
There is already an `optional` marking on the javadoc. Resolving.